### PR TITLE
Refactor FXIOS-7651 [v121] RustAutofill class cleanup

### PR DESCRIPTION
**Pull request by jnrahme**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-7651)

## :bulb: Description
Refactored class removed repeated code, put RusAutofillEncriptionKeys and UnencryptedCreditCardFields in seperate files 

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

**Comment by lmarceau** [comment]
FYI the build failed since `RustAutofillTests` couldn't compile

File: Storage/Rust/RustAutofill.swift [hunk 1]
```diff
@@ -524,9 +244,59 @@ public class RustAutofill {
             // but is disallowed nonetheless
             throw AutofillEncryptionKeyError.illegalState
         }
-
         // This must be declared again for Swift's sake even though the above switch statement
         // handles all cases
         throw AutofillEncryptionKeyError.illegalState
     }
+
+    // Private Helper Methods
```
Comment (adudenamedruby) [inline Storage/Rust/RustAutofill.swift:252]:
```suggestion
    // MARK: - Private Helper Methods
```

File: Storage/Rust/RustAutofill.swift [hunk 2]
```diff
@@ -9,296 +9,68 @@ import Common
...
         self.databasePath = databasePath
-
-        queue = DispatchQueue(label: "RustAutofill queue: \(databasePath)",
-                              attributes: [])
+        queue = DispatchQueue(label: "RustAutofill queue: \(databasePath)")
         self.logger = logger
     }
 
-    private func open() -> NSError? {
+    public func open() -> NSError? {
```
Comment (lmarceau) [inline Storage/Rust/RustAutofill.swift:32]: Can be internal I believe rather than public?

File: Storage/Rust/RustAutofill.swift [hunk 3]
```diff
@@ -9,296 +9,68 @@ import Common
...
-                return open()
-            }
-
+            handleDatabaseError(err)
             return err
         }
     }
 
-    private func close() -> NSError? {
+    public func close() -> NSError? {
```
Comment (lmarceau) [inline Storage/Rust/RustAutofill.swift:44]: Same, can be internal?

File: Storage/Rust/RustAutofill.swift [hunk 4]
```diff
@@ -308,42 +80,20 @@ public class RustAutofill {
         }
     }
 
-    public func getCreditCard(id: String, completion: @escaping (CreditCard?, Error?) -> Void) {
```
Comment (lmarceau) [inline Storage/Rust/RustAutofill.swift:311]: Wasn't used right?

**Comment by adudenamedruby** [comment]
Is this ready for re-review?

**Comment by lmarceau** [comment]
@jnrahme There's still two small nits from me (not sure if Roux still had things pending), but wondering if you saw them or not.

**Review by adudenamedruby — APPROVED** [review]

**Pull request merged**